pkg/svc/zserver.go

```diff
@@ -2,51 +2,206 @@ package svc
+	if req.GetDescription() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "Description can't be empty")
+	}
+	if req.GetService() != 1 {
+		conn, err := GRPCConnect()
+		if err != nil {
+			return nil, err
+		}
+		defer conn.Close()
+		client := rpb.NewMyResponderClient(conn)
+		resp, err := client.UpdateDescription(ctx, &rpb.UpdateDescriptionRequest{Description: req.Description, Service: req.Service})
+		if err != nil {
+			return nil, err
+		}
+		return &pb.UpdateDescriptionResponse{Description: resp.Description}, nil
+	}
+	s.IncRequests()
+	s.UpdateDescriptionFromServer(req.Description)
+	return &pb.UpdateDescriptionResponse{Description: s.GetDescriptionFromServer()}, nil
+}
+
+func (s *server) GetDescription(ctx context.Context, req *pb.GetDescriptionRequest) (*pb.GetDescriptionResponse, error) {
+	if req.GetService() != 1 {
+		conn, err := GRPCConnect()
+		if err != nil {
+			return nil, err
+		}
+		defer conn.Close()
+		client := rpb.NewMyResponderClient(conn)
+		resp, err := client.GetDescription(ctx, &rpb.GetDescriptionRequest{Service: req.Service})
+		if err != nil {
+			return nil, err
+		}
+		return &pb.GetDescriptionResponse{Description: resp.Description}, nil
+
+	}
+	s.IncRequests()
+	return &pb.GetDescriptionResponse{Description: s.GetDescriptionFromServer()}, nil
+}
+
+// Returns uptime in seconds
+func (s *server) GetUptime(ctx context.Context, req *pb.GetUptimeRequest) (*pb.GetUptimeResponse, error) {
+	if req.GetService() != 1 {
+		conn, err := GRPCConnect()
+		if err != nil {
+			return nil, err
+		}
+		defer conn.Close()
+		client := rpb.NewMyResponderClient(conn)
+		resp, err := client.GetUptime(ctx, &rpb.GetUptimeRequest{Service: req.Service})
+		if err != nil {
+			return nil, err
+		}
+		return &pb.GetUptimeResponse{Uptime: resp.Uptime}, nil
+
+	}
+	s.IncRequests()
+	uptime := time.Now().Unix() - s.Timestamp.Unix()
+	return &pb.GetUptimeResponse{Uptime: uptime}, nil
+}
+
+func (s *server) GetRequests(ctx context.Context, req *pb.GetRequestsRequest) (*pb.GetRequestsResponse, error) {
+	if req.GetService() != 1 {
+		conn, err := GRPCConnect()
+		if err != nil {
+			return nil, err
+		}
+		defer conn.Close()
+		client := rpb.NewMyResponderClient(conn)
+		resp, err := client.GetRequests(ctx, &rpb.GetRequestsRequest{Service: req.Service})
+		if err != nil {
+			return nil, err
+		}
+		return &pb.GetRequestsResponse{Requests: resp.Requests}, nil
+
+	}
+	s.IncRequests()
+	return &pb.GetRequestsResponse{Requests: int64(s.GetRequestsFromServer())}, nil
+}
+
+func (s *server) GetMode(ctx context.Context, req *pb.GetModeRequest) (*pb.GetModeResponse, error) {
+	conn, err := GRPCConnect()
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+	client := rpb.NewMyResponderClient(conn)
+	resp, err := client.GetMode(ctx, &rpb.GetModeRequest{})
+	if err != nil {
+		return nil, err
+	}
+	s.IncRequests()
```

Comment:
As I can see, in some cases we increase requests counter only when request is for Portal, but here it would be increased even if it was for Storage or Responder. Please use the same approach everywhere.